examples/mnist.py

```diff
@@ -30,6 +32,11 @@ def main(save_to, num_epochs):
     cost = CategoricalCrossEntropy().apply(y.flatten(), probs)
     error_rate = MisclassficationRate().apply(y.flatten(), probs)
 
+    cg = ComputationGraph([cost])
+    W1, W2 = VariableFilter(roles=[VariableRole.WEIGHTS])(cg.variables)
```

Comment:
I really like your idea about returning variables in topological order! This piece of code looks like magic.